File: app/src/main/java/io/github/kutaiba_nezar_kashmar/multiverse_of_geeks/domain/response/media/tv_responses/SingleTvShowResponse.java
package io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.domain.response.media.tv_responses;

import java.util.ArrayList;

import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.domain.response.media.MediaGenreResponse;
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.domain.response.media.MediaProductionCompaniesResponse;
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.domain.response.media.MediaProductionCountriesResponse;
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.domain.response.media.MediaSpokenLanguagesResponse;

public class SingleTvShowResponse
{
  private boolean adult;
  private String backdrop_path;
  private ArrayList<TvShowCreatorResponse> created_by;
  private ArrayList<Integer> episode_run_time;
  private String first_air_date;
  private ArrayList<MediaGenreResponse> genres;
  private String homepage;
  private int id;
  private boolean in_production;
  private ArrayList<String> languages;
  private String last_air_date;
  private TvShowLastEpisodeToAirResponse last_episode_to_air;
  private String name;
  private ArrayList<TvShowNetworkResponse> networks;
  private int number_of_episodes;
  private int number_of_seasons;
  private ArrayList<String> origin_country;
  private String original_language;
  private String original_name;
  private String overview;
  private float popularity;
  private String poster_path;
  private ArrayList<MediaProductionCompaniesResponse> production_companies;
  private ArrayList<MediaProductionCountriesResponse> production_countries;
  private ArrayList<TvShowSeasonResponse> seasons;
  private ArrayList<MediaSpokenLanguagesResponse> spoken_languages;
  private String status;
  private String tagline;
  private String type;
  private float vote_average;
  private float vote_count;

  public boolean isAdult()
  {
    return adult;
  }

  public String getBackdrop_path()
  {
    return backdrop_path;
  }

  public ArrayList<TvShowCreatorResponse> getCreated_by()
  {
    return created_by;
  }

  public ArrayList<Integer> getEpisode_run_time()
  {
    return episode_run_time;
  }

  public String getFirst_air_date()
  {
    return first_air_date;
  }

  public ArrayList<MediaGenreResponse> getGenres()
  {
    return genres;
  }

  public String getHomepage()
  {
    return homepage;
  }

  public int getId()
  {
    return id;
  }

  public boolean isIn_production()
  {
    return in_production;
  }

  public ArrayList<String> getLanguages()
  {
    return languages;
  }

  public String getLast_air_date()
  {
    return last_air_date;
  }

  public TvShowLastEpisodeToAirResponse getLast_episode_to_air()
  {
    return last_episode_to_air;
  }

  public String getName()
  {
    return name;
  }

  public ArrayList<TvShowNetworkResponse> getNetworks()
  {
    return networks;
  }

  public int getNumber_of_episodes()
  {
    return number_of_episodes;
  }

  public int getNumber_of_seasons()
  {
    return number_of_seasons;
  }

  public ArrayList<String> getOrigin_country()
  {
    return origin_country;
  }

  public String getOriginal_language()
  {
    return original_language;
  }

  public String getOriginal_name()
  {
    return original_name;
  }

  public String getOverview()
  {
    return overview;
  }

  public float getPopularity()
  {
    return popularity;
  }

  public String getPoster_path()
  {
    return poster_path;
  }

  public ArrayList<MediaProductionCompaniesResponse> getProduction_companies()
  {
    return production_companies;
  }

  public ArrayList<MediaProductionCountriesResponse> getProduction_countries()
  {
    return production_countries;
  }

  public ArrayList<TvShowSeasonResponse> getSeasons()
  {
    return seasons;
  }

  public ArrayList<MediaSpokenLanguagesResponse> getSpoken_languages()
  {
    return spoken_languages;
  }

  public String getStatus()
  {
    return status;
  }

  public String getTagline()
  {
    return tagline;
  }

  public String getType()
  {
    return type;
  }

  public float getVote_average()
  {
    return vote_average;
  }

  public float getVote_count()
  {
    return vote_count;
  }
}

File: app/src/main/java/io/github/kutaiba_nezar_kashmar/multiverse_of_geeks/model/domain/response/media/tv_responses/SingleTvShowResponse.kt
package io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response.media.tv_responses

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response.media.MediaGenreResponse
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response.media.MediaProductionCompaniesResponse
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response.media.MediaProductionCountriesResponse
import io.github.kutaiba_nezar_kashmar.multiverse_of_geeks.model.domain.response.media.MediaSpokenLanguagesResponse
import java.util.ArrayList

@Entity(tableName = "tv")
data class SingleTvShowResponse
    (
        var backdrop_path: String,
        var first_air_date: String,
        var homepage: String,
        @PrimaryKey var id: Int,
        var in_production: Boolean,
        var last_air_date: String,
        var name: String,
        var number_of_episodes: Int,
        var number_of_seasons: Int,
        var original_language: String,
        var original_name: String,
        var overview: String,
        var popularity: Float,
        var poster_path: String,
        var status: String,
        var tagline: String,
        var type: String,
        var vote_average: Float,
        var vote_count: Float,
)
{
    @Ignore
    val adult: Boolean = false

    @Ignore
    val created_by: List<TvShowCreatorResponse> = ArrayList()

    @Ignore
    val episode_run_time: List<Integer> = ArrayList()

    @Ignore
    val genres: List<MediaGenreResponse> = ArrayList()

    @Ignore
    val languages: List<String> = ArrayList()

    @Ignore
    val last_episode_to_air: TvShowLastEpisodeToAirResponse =
        TvShowLastEpisodeToAirResponse()

    @Ignore
    val networks: List<TvShowNetworkResponse> = ArrayList()

    @Ignore
    val origin_country: List<String> = ArrayList()

    @Ignore
    val production_companies: List<MediaProductionCompaniesResponse> =
        ArrayList()

    @Ignore
    val production_countries: List<MediaProductionCountriesResponse> =
        ArrayList()

    @Ignore
    val seasons: List<TvShowSeasonResponse> = ArrayList()

    @Ignore
    val spoken_languages: List<MediaSpokenLanguagesResponse> = ArrayList()
}